devito/core/autotuning.py

```diff
@@ -250,7 +250,7 @@ def finalize_time_bounds(stepper, at_args, args, mode):
 
 
 def calculate_nblocks(tree, blockable):
-    collapsed = tree[:(tree[0].ncollapsed or 1)]
```

Comment:
why "1"? this smells like a hack. Or, at least, it's dirty

Reply:
what makes you sure that tree[0] is time here? (and btw, what so special about time?)

could it instead have been:
```
for t
  for f
    for xb
      for yb
        ...
```

so two loops before the block loops xb/yb?

Reply:
this is unanswered